app/entity-repositories/pasteRepository.js

```diff
@@ -0,0 +1,101 @@
+class PasteRepository {
+    constructor ({ dbProvider }) {
+        this.dbProvider = dbProvider;
+    }
+
+    async findByID({ id }) {
+
+        const result = await this.findOne({ id });
+        return result;
+
+    }
+
+    async findOne({ id, name, authorID }) {
+        
+        const result = await this.findAll({ id, name, authorID });
+        
+        if(!result) {
+            return null;
+        }
+
+        return result[0];
+
+    }
+
+    async findAll({ id, name, authorID}) {
+
+        const itemsToFind = [];
+
+        if ( id ) {
+
+            const arrayOfIDs = [];
+
+            if (Array.isArray(id)) {
+                
+                for (let i = 0; i < id.length; i++) {
+                    arrayOfIDs.push(id[i]);
+                }
+
+                itemsToFind.push(`id in (${arrayOfIDs.join(', ')})`);
+
+            } else {
+
+                itemsToFind.push(`id in (${id})`);
+
+            }
```

Comment:
If you threat `id` as array (using SQL `in ()` construction) than rename the argument from `id` to `ids`. This forces us always to pass `id` as an array, so we don't care about is it actually one or more ids are passed.